core/templates/components/state-directives/solution-editor/solution-editor.directive.ts

```diff
@@ -74,10 +74,28 @@ angular.module('oppia').directive('solutionEditor', [
         function(StateSolutionService) {
           var ctrl = this;
           ctrl.getAnswerHtml = function() {
-            return ExplorationHtmlFormatterService.getAnswerHtml(
-              StateSolutionService.savedMemento.correctAnswer,
-              StateInteractionIdService.savedMemento,
-              StateCustomizationArgsService.savedMemento);
+            var interactionId = StateInteractionIdService.savedMemento;
+            if (interactionId === 'DragAndDropSortInput') {
+              var answer = ctrl.getDragAndDropSolution();
+              return ExplorationHtmlFormatterService.getAnswerHtml(
+                answer,
+                StateInteractionIdService.savedMemento,
+                StateCustomizationArgsService.savedMemento);
+            } else {
+              return ExplorationHtmlFormatterService.getAnswerHtml(
+                StateSolutionService.savedMemento.correctAnswer,
+                StateInteractionIdService.savedMemento,
+                StateCustomizationArgsService.savedMemento);
+            }
+          };
+
+          ctrl.getDragAndDropSolution = function() {
+            var solution = (StateSolutionService.savedMemento);
+            var solutionAsPlainText =
+              solution.getDragAndDropAnswer(
+                StateInteractionIdService.savedMemento,
+                StateCustomizationArgsService.savedMemento);
+            return solutionAsPlainText;
```

Comment:
Why we need the special case here? Can this be solved in some more generic way?

Reply:
The function getDragAndDropSolution() is for choices which user types instead of ca_choices_0, and so on. Here we can't use getSummary function because it returns the choices and the explanation, but here we only need to display the choices. So for that we a separate function. Here in getDragAndDropSolution function we are only returning the choices.

Reply:
And we need this function here because "StateSolutionService.savedMemento.correctAnswer" is not what we want, we want choices instead so we will need to get it from somewhere to proceed further. I hope this answers your question.